source/features/restore-file.tsx

```diff
@@ -76,30 +76,19 @@ async function restoreFile(progress: (message: string) => void, menuItem: Elemen
 	}
 }
 
-const filesRestored = new WeakSet<HTMLButtonElement>();
 async function handleRestoreFileClick(event: delegate.Event<MouseEvent, HTMLButtonElement>): Promise<void> {
 	const menuItem = event.delegateTarget;
 
-	// Only allow one click
-	if (filesRestored.has(menuItem)) {
-		return;
-	}
-
-	filesRestored.add(menuItem);
-
-	try {
+	await showToast(async progress => {
```

Comment:
`showToast` already acts as its own try/catch, with errors being shown in the toast itself. No need for an extra try/catch as long as everything is inside.